index.bs

```diff
@@ -560,79 +560,251 @@ urlPrefix: https://w3c.github.io/FileAPI/#dfn-; type: dfn;
+		</p>
+
+		<h4 attribute for=ClipboardItem lt=types>types</h4>
+		<p>
+			The {{ClipboardItem/types}} getter steps are to return [=this=]'s [=ClipboardItem/types array=].
+		</p>
+
+		<h4 method for=ClipboardItem lt=getType(type)>getType(type)</h4>
+		
+			This method must run the below steps:
+
+			1. Let |realm| be [=this=]'s [=relevant realm=].
+
+			1. Let |mimeType| be the result of [=parsing a MIME type=] given |type|.
+
+			1. If |mimeType| is failure, then throw a {{TypeError}}.
+
+			1. Let |itemTypeList| be [=this=]'s [=ClipboardItem/clipboard item=]'s [=list of representations=].
 
-		enum PresentationStyle { "unspecified", "inline", "attachment" };
+			1. Let |p| be [=a new promise=] in |realm|.
 
-		dictionary ClipboardItemOptions {
-		  PresentationStyle presentationStyle = "unspecified";
+			1. For each |representation| in |itemTypeList|:
+
+				1. If |representation|'s [=representation/MIME type=] is |mimeType|, then:
+
+					1. Let |representationDataPromise| be the |representation|'s [=representation/data=].
+
+					1. [=promise/React=] to |representationDataPromise|:
+
+						1. If |representationDataPromise| was fulfilled with value |v|, then:
+
+							1. If |v| is a {{DOMString}}, then follow the below steps:
+
+								1. Let |dataAsBytes| be the result of [=UTF-8 encoding=] |v|.
+
+								1. Let |blobData| be a {{Blob}} created using |dataAsBytes| with its {{Blob/type}} set to |mimeType|, [=serialize a MIME type|serialized=].
+
+								1. Resolve |p| with |blobData|.
+
+							1. If |v| is a {{Blob}}, then follow the below steps:
+
+								1. Resolve |p| with |v|.
+
+						1. If |representationDataPromise| was rejected, then:
+
+							1. [=Reject=] |p| with {{"NotFoundError"}} {{DOMException}} in |realm|.
+
+					1. Return |p|.
+
+			1. [=Reject=] |p| with {{"NotFoundError"}} {{DOMException}} in |realm|.
+
+			1. Return |p|.
+
+	<h3 id="clipboard-interface">Clipboard Interface</h3>
+
+		<pre class="idl" data-highlight="webidl">
+		typedef sequence&lt;ClipboardItem&gt; ClipboardItems;
+
+		[SecureContext, Exposed=Window]
+		interface Clipboard : EventTarget {
+			Promise&lt;ClipboardItems&gt; read();
+			Promise&lt;DOMString&gt; readText();
+			Promise&lt;undefined&gt; write(ClipboardItems data);
+			Promise&lt;undefined&gt; writeText(DOMString data);
 		};
 		</pre>
 
-		<div id="clipboard-idl" dfn-for="Clipboard">
+		The methods of the {{Clipboard}} interface take or return multiple {{ClipboardItem}} objects, and their specification is written to deal with the generic case. However, not all platforms support more than one [=/clipboard item=]; on such platforms, the algorithms below will ignore any {{ClipboardItem}} objects beyond the first one that are passed to {{Clipboard/write()}}, and {{Clipboard/read()}} and {{Clipboard/readText()}} will only ever return a single-item array.
```

Comment:
`Some methods` is more accurate.
```, and their specification is written to deal with the generic case.``` could be dropped, it's redundant.

```
on such platforms, the algorithms below will ignore any {{ClipboardItem}} objects beyond the first one that are passed to {{Clipboard/write()}}, and {{Clipboard/read()}} and {{Clipboard/readText()}} will only ever return a single-item array.
```
isn't entirely true, please see https://github.com/w3c/clipboard-apis/pull/158#discussion_r786017821. The simplest fix would be to change the above cited text to a note with text like:
```
Some methods of the {{Clipboard}} interface take or return multiple {{ClipboardItem}} objects. However, not all platforms support more than one [=/clipboard item=]; on such platforms: `write()` writes all items in sequence, hence the last one replaces the previous ones.
```
The methods `read()` and `readText()` automatically behave correct, since there's only one ClipboardItem.

However, the algorithm of `readText()` needs to be corrected for the case when there are **multiple** clipboard items on the system clipboard, each containing a "text/plain" representation.

Reply:
> isn't entirely true, please see #158 (comment).

In `write()`, the algorithm has a step to deal with multiple `clipboard item`s: 
`1. If |data|'s [=list/size=] is greater than 1, and the current operating system does not support multiple [=/clipboard item=]s on the [=system clipboard data=], then set |data| to |data|[0].`

FWIW, we've been discussing this internally whether we will ever support this use case or not, and we came to a conclusion that since this is very specific to a particular platform (macOS, iPadOS etc) we probably won't implement this in the near future. So, to limit the scope of this PR, I'll add text to just return the first `clipboard item` in `readText()` if the platform doesn't support it. 

However, I'll let Apple devs clarify this algorithm in a separate PR if they want to add some texts to these methods, since they are the only ones who know how multiple `clipboard item`s would behave in these methods.

Reply:
Thanks. It makes sense to limit the scope of this PR.